boards/arm/stm32f7/stm32f777zit6-meadow/src/Makefile

```diff
@@ -57,4 +59,120 @@ else ifeq ($(CONFIG_STM32F7_OTGFSHS),y)
 CSRCS += stm32_usb.c
 endif
 
+ifeq ($(CONFIG_USBMSC),y)
+CSRCS += stm32_usbmsc.c
+endif
+
+ifeq ($(CONFIG_ARCH_IDLE_CUSTOM),y)
+CSRCS += stm32_idle.c
+endif
+
+ifeq ($(CONFIG_STM32F7_SDMMC2),y)
+CSRCS += meadow_sdmmc.c
+endif
+
+ifeq ($(CONFIG_MEADOW_HCOM),y)
+CSRCS += $(MEADOW_OS_DIR)/hcom_nx/hcom_nx_upd.c \
+$(MEADOW_OS_DIR)/hcom_nx/hcom_nx_startup_mgr.c \
+$(MEADOW_OS_DIR)/hcom_nx/hcom_nx_common_utils.c \
+$(MEADOW_OS_DIR)/hcom_nx/hcom_nx_bbreg_access.c \
+$(MEADOW_OS_DIR)/hcom_nx/hcom_nx_upd_route_cmd.c \
+$(MEADOW_OS_DIR)/hcom_nx/hcom_nx_config_manager.c \
+$(MEADOW_OS_DIR)/hcom_nx/commands/hcom_nx_ex_flash.c \
+$(MEADOW_OS_DIR)/hcom_nx/comms/hcom_nx_host_send.c \
+$(MEADOW_OS_DIR)/hcom_nx/create_fs/hcom_nx_fs.c \
+$(MEADOW_OS_DIR)/hcom_nx/create_fs/hcom_nx_fs_littlefs.c \
+$(MEADOW_OS_DIR)/hcom_nx/diag/hcom_nx_route_text_to_host.c \
+$(MEADOW_OS_DIR)/hcom_nx/diag/hcom_nx_trace_msg_proc.c \
+$(MEADOW_OS_DIR)/hcom_nx/tests/hcom_nx_qspi_flash_tests.c
+
+ifeq ($(CONFIG_ARCH_IDLE_CUSTOM),y)
+CSRCS +=  misc/meadow_idle_monitor.c
+endif
+
+ifeq ($(CONFIG_MEADOW_PWR_MGMT_SUPPORT),y)
+CSRCS += pwrmgmt/pwrmgmt_calib_lsi_clock.c \
+pwrmgmt/pwrmgmt_config_wakeup_timer.c \
+pwrmgmt/pwrmgmt_config_wakeup_alarm.c \
+pwrmgmt/pwrmgmt_control.c \
+pwrmgmt/pwrmgmt_enter_stop_mode.c \
+pwrmgmt/pwrmgmt_misc.c \
+pwrmgmt/pwrmgmt_switch_rtc_clock.c \
+pwrmgmt/pwrmgmt_time_support.c
+endif
+
+ifeq ($(CONFIG_MEADOW_TIMER_SUPPORT),y)
+CSRCS += timers/timer_manager.c \
+timers/cpu_measure.c \
+timers/pulse_width.c \
+timers/freq_duty.c \
+timers/rc_servo_decode.c \
+timers/lsi_clk_freq_measure.c
+endif
+
+CSRCS += $(MEADOW_OS_DIR)/ethernet/meadow_ethnet_start.c \
```

Comment:
look like that you move many userspace components to kernel, it isn't a good design. @patacongo what do you thinking?

Reply:
@xiaoxiang781216 currently their implementation uses many functions in kernel mode that could be in userspace. So their final goal is having everything separated in kernel space, but before that they want start using the mainline to make the transition more smooth. All his modifications are board specific and doesn't compromise the NuttX kernel, since it doesn't apply to any other board.

Reply:
Can we give some kind of note in the code about what Xiao and Alan said? Something about that this is a work in progress design and that the userspace-kernel separation will be consistently improved.

Reply:
Yes, since the most change inside the board specific folder, it's fine to improve the separation step by step.

Reply:
That's correct, we hope that our board-specific patch is okay to merge as-is, as an integration patch to mainline NuttX from an older NuttX and not as an architecture enhancement patch.

Reply:
@raiden00pl @xiaoxiang781216 I have added notes in the Makefile + commit message marking this board work as under development. I have also updated the PR description to make it clearer there too.

Thank you!